File: cis_interface/examples/python/rpcFib/rpcFibSrv.py
from __future__ import print_function
import os
import sys
import time
import logging
from logging import debug, info
from cis_interface.interface.PsiInterface import *
import socket


def fibServer(sleeptime):
    print('fibsrv(P): hello on system {} with sleep {}'.format(
        socket.gethostname(), sleeptime))

    rpc = PsiRpc("srv_fib", "%d %d", "srv_fib", "%d")

    while True:
        print('fibsrv(P): input<- ', end='')
        retval, rpc_in = rpc.rpcRecv()
        if not retval:
            print('rpcFibSrv: end of input')
            break

        #
        # compute fibonacci(input)
        #
        pprev = 0
        prev = 1
        result = 1
        fib_no = 1
        arg = rpc_in[0]
        while fib_no < arg:
            result = prev + pprev
            pprev = prev
            prev = result
            fib_no = fib_no+1
        print(' ::: ->(%2d %2d)' % (arg, result))

        time.sleep(float(sleeptime))
        rpc.rpcSend(arg, result)

    debug('fibsrv(P): bye')
    sys.exit(0)

if __name__ == '__main__':
    LOGLEVEL = logging.NOTSET
    if 'PSI_CLIENT_DEBUG' in os.environ:
        LOGLEVEL = getattr(logging, os.environ['PSI_CLIENT_DEBUG'])
    if 'RMQ_DrEBUG' in os.environ:
        RMQLOGLEVEL = getattr(logging, os.environ['RMQ_DEBUG'])
    logging.basicConfig(level=LOGLEVEL, stream=sys.stdout, \
	format=sys.argv[0].split('/')[-1]+': %(message)s')
    print('psirun', sys.argv)
    fibServer(sys.argv[1])

File: cis_interface/examples/matlab/rpcFib/rpcFibSrv.m

function rpcFibSrv(sleeptime)
  PsiInterface = py.importlib.import_module('cis_interface.interface.PsiInterface');
  sleeptime = str2num(sleeptime);
  
  disp('rpcFibSrv starts')

  rpc = PsiInterface.PsiRpc('srv_fib', '%d %d', 'srv_fib', '%d');

  while 1
    input = rpc.rpcRecv();
    if (~input{1})
      D = sprintf('rpcFibSrv(M): end of input');
      disp(D);
      exit(0);
    end
    input = input{2};
    D = sprintf('rpcFibSrv(M): got input %d', input{1});
    disp(D);

    pprev = 0;
    prev = 1;
    result = 1;
    n = 1;
    while n < input{1}
      result = prev + pprev;
      pprev = prev;
      prev = result;
      n = n+1;
    end
    pause(sleeptime);
    D = sprintf('rpcFibSrv(M): sending result %d %d', input{1}, result);
    disp(D);
    rpc.rpcSend(input{1}, result);
  end

  disp('rpcFibSrv: done')
  exit(0);
end



